src/controllers/queryRunner.ts

```diff
@@ -280,42 +294,42 @@ export default class QueryRunner {
     /*
      * Get more data rows from the current resultSets from the service layer
      */
-    public getRows(rowStart: number, numberOfRows: number, batchIndex: number, resultSetIndex: number): Promise<QueryExecuteSubsetResult> {
-        const self = this;
+    public async getRows(rowStart: number, numberOfRows: number, batchIndex: number, resultSetIndex: number): Promise<QueryExecuteSubsetResult> {
         let queryDetails = new QueryExecuteSubsetParams();
         queryDetails.ownerUri = this.uri;
         queryDetails.resultSetIndex = resultSetIndex;
         queryDetails.rowsCount = numberOfRows;
         queryDetails.rowsStartIndex = rowStart;
         queryDetails.batchIndex = batchIndex;
-        return new Promise<QueryExecuteSubsetResult>((resolve, reject) => {
-            self._client.sendRequest(QueryExecuteSubsetRequest.type, queryDetails).then(result => {
-                resolve(result);
-            }, error => {
-                // TODO: Localize
-                self._vscodeWrapper.showErrorMessage('Something went wrong getting more rows: ' + error.message);
-                reject();
-            });
-        });
+        try {
+            const queryExecuteSubsetResult = await this._client.sendRequest(QueryExecuteSubsetRequest.type, queryDetails);
+            if (queryExecuteSubsetResult) {
+                return queryExecuteSubsetResult;
+            }
+        } catch (error) {
+            // TODO: Localize
+            this._vscodeWrapper.showErrorMessage('Something went wrong getting more rows: ' + error.message);
```

Comment:
>Something went wrong getting more rows [](start = 50, length = 38)

This error message isn't great, would suggest changing it to 'Unexpected error occurred retrieving more rows: ' + error.message.